crates/nargo/src/cli/check_cmd.rs

```diff
@@ -63,11 +64,26 @@ fn check_from_path<P: AsRef<Path>>(p: P, allow_warnings: bool) -> Result<(), Cli
     Ok(())
 }
 
-fn build_empty_map(abi: Abi) -> BTreeMap<String, &'static str> {
-    btree_map(abi.parameters, |AbiParameter { name, typ, .. }| {
-        let default_value = if matches!(typ, AbiType::Array { .. }) { "[]" } else { "" };
-        (name, default_value)
-    })
+fn build_empty_map(
+    parameters: Vec<AbiParameter>,
+    return_type: Option<AbiType>,
+) -> BTreeMap<String, &'static str> {
+    let default_value = |typ: AbiType| {
+        if matches!(typ, AbiType::Array { .. }) {
+            "[]"
+        } else {
+            ""
+        }
+    };
+
+    let mut map =
+        btree_map(parameters, |AbiParameter { name, typ, .. }| (name, default_value(typ)));
```

Comment:
If we're already factoring out the lambda, I think we should go all the way and include the whole lambda in `default_value`.

Reply:
This gets a bit mucky when it comes to dealing with the return value however. We'd have to pass in a dummy name just to immediately extract the default value term. This seems cleaner imo.